### PR TITLE
remove curve25519_64_slow from dist

### PR DESCRIPTION
**Pull request by polubelova**
This PR removes `Hacl_Curve25519_64_Slow` from `dist/gcc-compatible` as this variant was used only for proofs (needed for the bignum library using a 2^64-radix representation).

### CODE REVIEW

**Comment by polubelova** [comment]
I don't remember this discussion. I think it's better to remove `Hacl_Curve25519_64_Slow` to ensure that it's not used by chance. We have a portable C implementation (`Hacl_Curve25519_51`) that uses a 2^51-radix representation and  a mixed assembly-C implementation (`Hacl_Curve25519_64`) that uses a 2^64-radix representation and relies on the Intel ADX and BMI2 instructions. So, there is no point in keeping `Hacl_Curve25519_64_Slow` in `dist/`.

**Comment by karthikbhargavan** [comment]
There are two things of value in this code:
(1) the proof that the 64-bit arithmetic is correct. This is mainly a proof in pure F*.
(2) the low-star code that relies on intrinsics. This is currently too slow to use, but if one day, compilers start supporting ADX intrinsics, this code could become relevant.

I find even just the proofs for (1) to be useful, but if Marina feels that this code and its proofs are super-ceded by Bignum256 (e.g.) then removing it makes sense to me. I agree we should make sure the code is not accidentally used but the "_Slow" suffix should ensure that, no?

**Comment by polubelova** [comment]
This PR keeps both proofs and Low* code for a 2^64-radix representation but doesn't extract it into `dist/`.

As for the ADX/MULX instructions, we may need to add a new implementation of Bignum256 using the `_addcarryx_u64 ` and `_mulx_u64` intrinsics. (Our Low* implementation for `curve25519_64_slow` relies on Bignum256 except for the modular reduction, so there is no "extra" code.)

**Review by karthikbhargavan — APPROVED** [review]

File: code/curve25519/Makefile [hunk 1]
```diff
@@ -7,7 +7,7 @@ all: dist/curve64-ours.exe
 test: all
 	dist/curve64-ours.exe
 
-FSTAR_ROOTS = Hacl.Curve25519_64_Local.fst Hacl.Curve25519_64_Slow.fst Hacl.Curve25519_51.fst
```
[the remark this answers is not in the document]
Comment (polubelova) [reply code/curve25519/Makefile:10]: you mean [here](https://github.com/hacl-star/hacl-star/blob/main/Makefile.common#L96) we should replace
```
CURVE_BUNDLE_SLOW= \
 -bundle 'Hacl.Curve25519_64_Slow=Hacl.Impl.Curve25519.Field64.Hacl,Hacl.Spec.Curve25519,Hacl.Spec.Curve25519.\*'
```
with
```
CURVE_BUNDLE_SLOW= -bundle Hacl.Curve25519_64_Slow
```
?
If so, your fix is better.

[the remark this answers is not in the document]
Comment (msprotz) [reply code/curve25519/Makefile:10]: Yes, I think that will still extract it but not code-gen it with krml

**Pull request merged**